### PR TITLE
Create spell for the latest usd price of every token in dex.trades

### PR DESCRIPTION
**Pull request by bernatfp**
PR #3830 is blocked by missing dex.trades in DuneSQL so making a separate PR for a Spark alternative to unblock an internal workstream.

See https://github.com/duneanalytics/spellbook/pull/3830 for more details.

### CODE REVIEW

File: models/dex/dex_prices_latest_legacy.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{ config(
-        alias = alias('prices_latest'),
+        alias = alias('prices_latest', legacy_model=True),
```
Comment (jeff-dude) [inline models/dex/dex_prices_latest_legacy.sql:2]: this is needed for when we add to dunesql to help differentiate table names

Comment (bernatfp) [reply models/dex/dex_prices_latest_legacy.sql:2]: ah got it, thanks Jeff!

File: models/dex/dex_prices_latest_legacy.sql [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
         tags = ['legacy'],
-        partition_by = ['block_time'],
+        partition_by = ['block_month'],
```
Comment (jeff-dude) [inline models/dex/dex_prices_latest_legacy.sql:4]: we're looking to universally change partitions to monthly level, as our partitions are too small
fyi, partitioning on `block_time` is much too granular, at minimum should be daily (if massive dataset)

Comment (bernatfp) [reply models/dex/dex_prices_latest_legacy.sql:4]: ok good to know, taking note

File: models/dex/dex_prices_latest_legacy.sql [hunk 3]
```diff
@@ -30,27 +32,32 @@ FROM (
...
             token_bought_address AS token_address,
             token_bought_amount AS token_amount,
             token_bought_amount_raw AS token_amount_raw,
             amount_usd
         FROM {{ ref('dex_trades_legacy') }}
-        {% if is_incremental() %}
-        WHERE block_time >= date_trunc("day", now() - interval '1 week')
-        {% endif %}
+        WHERE
+            amount_usd > 0
```
Comment (jeff-dude) [inline models/dex/dex_prices_latest_legacy.sql:43]: moved this filter to the inner query, ideally more performant, hope that's okay

**Review by jeff-dude — APPROVED** [review]
looks good! let's get it included in next merge

**Pull request merged**